internal/operands/metrics/reconcile_test.go

```diff
@@ -54,14 +57,54 @@ var _ = Describe("Metrics operand", func() {
 			Logger:       log,
 			VersionCache: common.VersionCache{},
 		}
+	})
+
+	AfterEach(func() {
+		os.Unsetenv(runbookURLTemplateEnv)
+	})
 
+	It("should create metrics resources", func() {
 		_, err := operand.Reconcile(&request)
 		Expect(err).ToNot(HaveOccurred())
-		ExpectResourceExists(newPrometheusRule(namespace), request)
+
+		prometheusRule, err := newPrometheusRule(namespace)
+		Expect(err).ToNot(HaveOccurred())
+
+		ExpectResourceExists(prometheusRule, request)
 		ExpectResourceExists(newServiceMonitorCR(namespace), request)
 		ExpectResourceExists(newMonitoringClusterRole(), request)
 		ExpectResourceExists(newMonitoringClusterRoleBinding(), request)
 	})
+
+	DescribeTable("runbook URL template",
+		func(template string) {
+			if template != defaultRunbookURLTemplate {
+				os.Setenv(runbookURLTemplateEnv, template)
+			}
+
+			prometheusRule, err := newPrometheusRule(namespace)
+
+			if strings.Count(template, "%s") != 1 || strings.Count(template, "%") != 1 {
+				Expect(err).To(HaveOccurred())
+				return
+			}
+
+			Expect(err).ToNot(HaveOccurred())
+
+			for _, group := range prometheusRule.Spec.Groups {
+				for _, rule := range group.Rules {
+					if rule.Alert != "" {
+						if rule.Annotations["runbook_url"] != "" {
+							Expect(rule.Annotations["runbook_url"]).To(Equal(fmt.Sprintf(template, rule.Alert)))
+						}
+					}
+				}
+			}
+		},
+		Entry("should use the default template when no ENV variable is set", defaultRunbookURLTemplate),
+		Entry("should use the ENV variable when a valid value is set", "valid/runbookURL/template/%s"),
+		Entry("should throw an error when an invalid ENV variable value is set", "invalid/runbookURL/template/"),
```

Comment:
Add negative test case for another format specifier, e.g. `%d`?

Reply:
Added